### PR TITLE
use TotalScore summarized in NodePluginScores

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

use TotalScore summarized in NodePluginScores in `selectHost`

#### Which issue(s) this PR fixes:

Part of: https://github.com/kubernetes/kubernetes/issues/107556

Next action: Include the top 3 Node score details (each plugin/extender score) in the event message. 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by sanposhiho** [comment]
/cc @ahg-g 
ref: https://github.com/kubernetes/kubernetes/pull/108494

File: pkg/scheduler/schedule_one.go [hunk 1]
```diff
@@ -707,49 +703,63 @@ func prioritizeNodes(
...
+					nodename := (*prioritizedList)[i].Host
+					score := (*prioritizedList)[i].Score
 					if klogV.Enabled() {
-						klogV.InfoS("Extender scored node for pod", "pod", klog.KObj(pod), "extender", extenders[extIndex].Name(), "node", host, "score", score)
+						klogV.InfoS("Extender scored node for pod", "pod", klog.KObj(pod), "extender", extenders[extIndex].Name(), "node", nodename, "score", score)
 					}
-					combinedScores[host] += score * weight
+
+					finalscore := score * weight * (framework.MaxNodeScore / extenderv1.MaxExtenderPriority)
+					mu.Lock()
```
Comment (ahg-g) [inline pkg/scheduler/schedule_one.go:714]: I think it is less expensive to keep the lock outside the loop, we are not really enabling extra parallelism by moving it inside.

File: pkg/scheduler/schedule_one.go [hunk 2]
```diff
@@ -707,49 +703,63 @@ func prioritizeNodes(
...
-					host, score := (*prioritizedList)[i].Host, (*prioritizedList)[i].Score
+					nodename := (*prioritizedList)[i].Host
+					score := (*prioritizedList)[i].Score
 					if klogV.Enabled() {
-						klogV.InfoS("Extender scored node for pod", "pod", klog.KObj(pod), "extender", extenders[extIndex].Name(), "node", host, "score", score)
+						klogV.InfoS("Extender scored node for pod", "pod", klog.KObj(pod), "extender", extenders[extIndex].Name(), "node", nodename, "score", score)
 					}
-					combinedScores[host] += score * weight
+
+					finalscore := score * weight * (framework.MaxNodeScore / extenderv1.MaxExtenderPriority)
```
Comment (ahg-g) [inline pkg/scheduler/schedule_one.go:713]: pls keep the original comment.

File: pkg/scheduler/schedule_one.go [hunk 3]
```diff
@@ -707,49 +703,63 @@ func prioritizeNodes(
...
-				mu.Unlock()
 			}(i)
 		}
 		// wait for all go routines to finish
 		wg.Wait()
-		for i := range result {
-			// MaxExtenderPriority may diverge from the max priority used in the scheduler and defined by MaxNodeScore,
-			// therefore we need to scale the score returned by extenders to the score range used by the scheduler.
-			result[i].Score += combinedScores[result[i].Name] * (framework.MaxNodeScore / extenderv1.MaxExtenderPriority)
+		for i := range nodesScores {
```
Comment (ahg-g) [inline pkg/scheduler/schedule_one.go:732]: why do we need to iterate again, lets just directly add the score in the parallel loop above and get rid of allNodeExtendersScores all together.

Comment (sanposhiho) [reply pkg/scheduler/schedule_one.go]: True, I'm not sure why I needed allNodeExtendersScores. 🤔

**Comment by sanposhiho** [comment]
@ahg-g Fixed based on your suggestions. Please retake a look 🙏

File: pkg/scheduler/schedule_one_test.go [hunk 4]
```diff
@@ -2334,14 +2335,191 @@ func TestZeroRequest(t *testing.T) {
...
+			scheduler := Scheduler{
+				nodeInfoSnapshot:         snapshot,
+				percentageOfNodesToScore: schedulerapi.DefaultPercentageOfNodesToScore,
+			}
+
+			state := framework.NewCycleState()
+			_, _, err = scheduler.findNodesThatFitPod(ctx, fwk, state, test.pod)
+			if err != nil {
+				t.Fatalf("error filtering nodes: %+v", err)
+			}
```
Comment (ahg-g) [inline pkg/scheduler/schedule_one_test.go:2500]: do you really need to call the filters for this test?

Comment (sanposhiho) [reply pkg/scheduler/schedule_one_test.go]: No. Removed it 🙏

**Comment by sanposhiho** [comment]
@ahg-g Sorry for being late. Please retake a look 🙇

**Comment by sanposhiho** [comment]
Hm, `Test_prioritizeNodes` seems flaky, let me check 🤔

**Comment by ahg-g** [comment]
> Hm, `Test_prioritizeNodes` seems flaky, let me check🤔

What was the change to fix the flake?

**Comment by sanposhiho** [comment]
Restored `allNodeExtendersScores` in 3e1adc2359c08b9299fb69445d2cdfc7beadb035. 
Tell me if you have any other idea than `allNodeExtendersScores`. 🙏

**Comment by ahg-g** [comment]
ok, I guess the issue is that we can't lookup the node using its name from `nodeScores`

pls squash

**Comment by sanposhiho** [comment]
Squashed.

**Pull request merged**